### ISSUE TITLE
Global defaults

### ISSUE THREAD

**Issue by davidnknight** [outbrain/pure-angular-date-range-picker#19]
It would be good to be able to set/override global defaults. If you're familiar with uib, there are options to provide global defaults via providers. See https://angular-ui.github.io/bootstrap/#/tooltip, $uibTooltipProvider for example, or my code sample below.

Here's example usage:
```
;(function() {
    'use strict';

    angular
        .module('app', [
            ...
            'ui.bootstrap',
            'obDateRangePicker'
        ])
        .config(uibDefaults);

    function uibDefaults($uibTooltipProvider, $uibModalProvider) {
        $uibTooltipProvider.options({
            appendToBody: true,
            placement: 'top'
        });

        $uibModalProvider.options = {
            controllerAs: 'vm',
            size: 'md'
        };
    }
})();
```

It would be nice to pass something similar to, all optional of course, falling back to package defaults if not passed in the provider:
```
;(function() {
    'use strict';

    angular
        .module('app', [
            ...
            'obDateRangePicker'
        ])
        .config(obDateRangePickerDefaults);

    function obDateRangePickerDefaults($obDateRangePickerProvider) {
        $obDateRangePickerProvider.options({
            start: ...,
            end: ...,
            ranges: [],
            ... etc
        });
    }
})();
```

If this was possible, it would be more DRY as I'm finding that I'm adding a lot of the same attributes when using the date pickers and copy/pasting the same set of start/end, ranges and even the on-apply, I'd rather default that to `vm.setDateRange()`.

The order of config applied would be supplied by attribute, provider and then package.

**Comment by vlio20** [comment]
You are right, will try to add it to next release.

**Comment by vlio20** [comment]
@davidnknight , Started working on this item, you can see progress on the provider brunch.

**Comment by davidnknight** [comment]
Cool. Thanks for letting me know @vlio20, very much looking forward to this. :)

**Comment by vlio20** [comment]
@davidnknight , I have just released 0.9.1 version which includes providers for both components. Please check it out and let me know if there are any issues.

Also it would be nice to see some implementations of the date and date-range picker, can you please provide a link?

Thanks for opening an issue.
